### ISSUE TITLE
Aggregate Reports Are Not Being Built

### ISSUE THREAD

**Issue by alzimmermsft** [Azure/azure-sdk-for-java#11197]
Since the parent POM was restructured to strictly be a parent POM our aggregate reports are no longer generating. This is due to there being no single target `pom.xml` which contains a module list which would allow for an aggregate to be generated. Currently these are the aggregate reports no longer being generated:

- [ ] [Javadocs](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/apidocs/index.html)
- [ ] [Dependency Issue Report](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/pom.client.html)
- [ ] [Maven Site](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/index.html)
- [ ] [Checkstyle Aggregate](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/checkstyle-aggregate.html)

**Comment by mitchdenny** [comment]
> This is due to there being no single target `pom.xml` which contains a module list which would allow for an aggregate to be generated. Currently these are the aggregate reports no longer being generated

Is this the case for all of the reports listed above. TBH the aggregate reporting support seems to be very inconsistent across the various tools we have.

**Comment by mitchdenny** [comment]
My thinking is that we'd rename the ```jacoco-test-coverage``` module to ```report-aggregator``` and have it be the module that we target all our aggregate reporting requirements ... the way it works with JaCoCo.

**Comment by alzimmermsft** [comment]
> 
> 
> My thinking is that we'd rename the `jacoco-test-coverage` module to `report-aggregator` and have it be the module that we target all our aggregate reporting requirements ... the way it works with JaCoCo.

I think this is a good direction to go in as it also simplifies all aggregate reporting to a single POM, meaning one less place to remember to update when adding a new module or updating a module.

**Comment by srnagar** [comment]
>Is this the case for all of the reports listed above.

Yes, all the reports listed above have not been updated. 

>My thinking is that we'd rename the jacoco-test-coverage module to report-aggregator and have it be the module that we target all our aggregate reporting requirements ... the way it works with JaCoCo.

I like this option. Do you think this will only be used for aggregate reporting or will any of the maven build options might be able to take advantage of this module with `-am` option? I just want to consider this before renaming it to `report-aggregator`.

**Comment by mitchdenny** [comment]
Actually at the moment, each time I look at this I get broken by failing/flakey tests in various modules :(

**Comment by JonathanGiles** [comment]
Really keen to get the aggregate whitelist dependency report generated, as well as all the other aggregate reports, ASAP. Being without this information makes me anxious.

**Comment by mitchdenny** [comment]
I'm assuming that we won't be retaining the term "whitelist". What do we want to change it to?

**Comment by JimSuplizio** [comment]
@mitchdenny @JonathanGiles @alzimmermsft @srnagar 
I tracked down why the allowlist report aren't being generated correctly and it's not due to the pom file restructuring work. [I filed an issue with the details](https://github.com/Azure/azure-sdk-for-java/issues/12535) but the short of it is that the whitelist generator needs a 1 line file and re-release. This should probably be done at the same time the [plugin is renamed](https://github.com/Azure/azure-sdk-for-java/issues/12530) as I've just merged a PR that changes whitelist to allowlist.
